aleph-client/src/pallets/multisig.rs

```diff
@@ -71,3 +129,388 @@ impl<S: SignedConnectionApi> MultisigUserApi for S {
+    ) -> AnyResult<(BlockHash, Option<Context>)> {
+        let other_signatories = ensure_signer_in_party(self, &context.party)?;
+
+        self.approve_as_multi(
+            context.party.threshold,
+            other_signatories,
+            Some(context.timepoint.clone()),
+            context.max_weight.clone(),
+            context.call_hash,
+            status,
+        )
+        .await
+        .map(|block_hash| {
+            (
+                block_hash,
+                context.add_approval(account_from_keypair(self.signer().signer())),
+            )
+        })
+    }
+
+    async fn approve_with_call(
+        &self,
+        mut context: Context,
+        call: Option<Call>,
+        status: TxStatus,
+    ) -> AnyResult<(BlockHash, Option<Context>)> {
+        let other_signatories = ensure_signer_in_party(self, &context.party)?;
+
+        let call = match (call.as_ref(), context.call.as_ref()) {
+            (None, None) => Err(anyhow!(
+                "Call wasn't provided earlier - you must pass it now"
+            )),
+            (None, Some(call)) => Ok(call),
+            (Some(call), None) => {
+                context.set_call(call)?;
+                Ok(call)
+            }
+            (Some(saved_call), Some(new_call)) => {
+                ensure!(
+                    saved_call == new_call,
+                    "The call is different that the one used previously"
+                );
+                Ok(new_call)
+            }
+        }?;
+
+        self.as_multi(
+            context.party.threshold,
+            other_signatories,
+            Some(context.timepoint.clone()),
+            context.max_weight.clone(),
+            call.clone(),
+            status,
+        )
+        .await
+        .map(|block_hash| {
+            (
+                block_hash,
+                context.add_approval(account_from_keypair(self.signer().signer())),
+            )
+        })
+    }
+
+    async fn cancel(&self, context: Context, status: TxStatus) -> AnyResult<BlockHash> {
+        let other_signatories = ensure_signer_in_party(self, &context.party)?;
+
+        let signer = account_from_keypair(self.signer().signer());
+        ensure!(
+            signer == context.author,
+            "Only the author can cancel multisig aggregation"
+        );
+
+        self.cancel_as_multi(
+            context.party.threshold,
+            other_signatories,
+            context.timepoint,
+            context.call_hash,
+            status,
+        )
+        .await
+    }
+}
+
+/// Compute hash of `call`.
+pub fn compute_call_hash(call: &Call) -> CallHash {
+    call.using_encoded(blake2_256)
+}
+
+/// Ensure that the signer of `conn` is present in `party.signatories`. If so, return all other
+/// signatories.
+fn ensure_signer_in_party<S: SignedConnectionApi>(
+    conn: &S,
+    party: &MultisigParty,
+) -> anyhow::Result<Vec<AccountId>> {
+    let signer_account = account_from_keypair(conn.signer().signer());
```

Comment:
isnt there something like conn.signer().account_id()

Reply:
perfect! I haven't noticed it before